base-isa.md

```diff
@@ -22,48 +23,45 @@ The highest two bits of the first byte are a format marker:
 | First byte    | Second Byte  | Comment                                  |
 |:--------------|:-------------|:-----------------------------------------|
 | `00 01 CCCC`  | `RRR RRR 00` | 2 register computation                   |
-| `00 SS CCCC`  | `RRR RRR ??` | when `SS != 01`, reserved for extensions |
-| `00 01 CCCC`  | `RRR RRR ??` | when `?? != 00`, reserved for extensions |
-| `00 01 11??`  | `RRR RRR ??` | reserved for extensions                  |
+| `00 01 11??`  | `???? ????`  | reserved for extensions                  |
+| `00 SS CCCC`  | `RRR RRR MM` | when `MM != 00`, reserved for extensions |
```

Comment:
GitHub won't let me annotate the whole range that I want because of the mix of +/- diff lines.

Aim to put general _legal_ patterns, followed by (possibly more than one) related patterns which are reserved, _from most general to most specific_. Overlap in those related patterns is fine; conceptually this is why most general should come first. It's easier to process. Then repeat with more legal patterns and more related reservations.

Rearrange so that we have 2 register computations, followed by the 2 register pattern with SS/MM. Specifically, from the latest commit, put line 27 after line 25. Then "when `MM != 00`" can be replaced by "when not matching above", which is more specific as it also covers the `SS` bits which are seemingly ignored here (they are mentioned below, but in a separate pattern that seems to be confusing overlap). 

Next put the current line 26, which explicitly reserves 4 opcodes so is less general than the previous pattern.

Then line 29 should remain `01 SS CCCC` (though the second byte => `???? ????` is good IMO). Similarly replace explanation with "when not matching above." The generalization of the `01` to `0?` helps compact the table but makes it harder to process.